### DISCUSSION TITLE
concourse_http_responses_duration_seconds for GetResource seems to always return 200

### DISCUSSION THREAD

**Discussion by gareth-e** [concourse/concourse#6146] [Help & Support] [answered]
Hi,
I've recently started interrogating emitted metrics in Prometheus, and was hoping to include failed resource checks on a dashboard.
However, it seems that a query such as `concourse_http_responses_duration_seconds_count{route="GetResource"}` always returns a status of 200 _(usually a good thing, you would think!)_
However, I've caused a few deliberate 4xx errors on resource checks (with invalid IDs, resource URIs, etc.) but I still only see 200 responses coming back in the metrics.
I've tried dropping the route from the query and added `{status="404"}` - then caused a few deliberate 404 errors on resource checks - still this metric does not show any increase.
Perhaps I have the wrong idea of what this metric actually does.
Does anyone have any clues or experience with this one?
Or a method to show failed resource checks from emitted metrics?
Thanks.

**Comment by gareth-e** [comment, marked as answer]
Oh, now then...
I'm thinking maybe what I'm looking for is....  `concourse_lidar_checks_finished_total{status="error"}`

**Comment by jamieklassen** [comment]
Off the top of my head, I would think the API action that would be erroring is `CheckResource` rather than `GetResource`

**Reply by gareth-e** [reply]
Hi @jamieklassen ,
Thanks for the info - yes, that one is better, though I think the `concourse_lidar_checks_finished_total{status="error"}` metric gives me more what I'm after. The `CheckResource` I see incrementing if I refresh a resource check though that always brings back a 201 whether the resource check is successful or otherwise (I guess the _request_ the check the resource worked - hence that response).
But the _"lidar_checks_finished status=error"_ metric I'm finding is useful for getting the actual status of the check itself - so I reckon I'll go with that one for this, but it's good to know that `CheckResource` route metric is there too.
Thanks.👍

**Reply by prime89-hash** [reply]
Hi @gareth-e , 

I need small help to understand how we can setup inbuilt concourse Prometheus metrics on aws cloudwatch dashboard.
Also it will be really helpful if you can give us an idea how exactly are you using these metrics in your implementation.

**Reply by gareth-e** [reply]
Hi @prime89-hash 
Actually, I don't work for the same company now and no longer use Concourse...
But, if it is some help... back then we were collecting using Prometheus (using the built-in Prometheus exporter in the web node) and visualising using Grafana dashboards.
You could try the AWS Prometheus Managed Service - see if with that you can read the emitted metrics and perhaps integrate that with a CloudWatch dashboard. I have to confess I've never tried that - but may be worth a go.
🤞
